### PR TITLE
feat: support struct response with @Failture

### PR DESCRIPTION
**Pull request by aecra**
#830

Just migrate the code from the @Success section into the @Failure section.

### CODE REVIEW

File: generate/swaggergen/g_docs.go [hunk 1]
```diff
@@ -695,23 +695,49 @@ func parserComments(f *ast.FuncDecl, controllerName, pkgpath string) error {
...
+						}
+					} else {
+						rs.Schema = &schema
+					}
+					rs.Description = strings.TrimSpace(ss[pos:])
+				} else {
+					rs.Description = strings.TrimSpace(ss)
 				}
-				opts.Responses[string(cd)] = rs
+				opts.Responses[respCode] = rs
```
Comment (flycash) [inline generate/swaggergen/g_docs.go:740]: can you help to extra this part to be a new function which returns the `swagger.Response`:
```
parseFailureResponse(annotation string) swagger.Response
```
And then you can write unit tests for this function.

I know this repository has no unit tests, but we can improve it

**Review by flycash — CHANGES_REQUESTED** [review]

File: generate/swaggergen/g_docs.go (hunk 1 above)
Comment (aecra) [reply generate/swaggergen/g_docs.go:740]: Turning this duplication into a standalone function would work, but with the current one-liner and shared memory model I don't think it's possible to write good unit tests.

Comment (flycash) [reply generate/swaggergen/g_docs.go]: It's not complicate.
You only need to reset the package variable when step in and step out the function, like this:
```go
func xxx() {
   reset()
    defer reset()
}
```

Comment (aecra) [reply generate/swaggergen/g_docs.go]: @flycash You can take a look at my new commit.

Comment (flycash) [reply generate/swaggergen/g_docs.go]: I check your commit and it looks good. But I didn't see any tests

Comment (aecra) [reply generate/swaggergen/g_docs.go]: I have never learn about package "go/ast", so I have no ieda to reset any variable.

Comment (aecra) [reply generate/swaggergen/g_docs.go]: If you think it is easy, you can finish this unit test.

**Pull request closed without merging**